airbyte-workers/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -229,6 +231,14 @@ private void handleEmittedErrorTrace(final AirbyteTraceMessage errorTraceMessage
     }
   }
 
+  private void handleEmittedNotificationTrace(final AirbyteTraceMessage notificationTraceMessage, final ConnectorType connectorType) {
+    final AirbyteNotificationTraceMessage notification = notificationTraceMessage.getNotification();
+    final String logMessage =
+        "[" + connectorType + " NOTIFICATION] " + notification.getMessage()
+            + notification.getData() != null ? " -> " + Jsons.serialize(notification.getData()) : "";
```

Comment:
In this case strings concatination will be done before null check because "+" has higher precedence then "?". And `"[" + connectorType + " NOTIFICATION] " + notification.getMessage()`  won't be part of logMessage.
Could you please replace this with:
```suggestion
            + (notification.getData() != null ? " -> " + Jsons.serialize(notification.getData()) : "");
```